# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.h
# Commit: 3ca24e761cc3
# Full Hash: 3ca24e761cc31bbb71c659661874a995ef3d3db1
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 6
# Description:
#   Bug 1654112 - Remove WebrtcGmpPCHandleSetter and plumb pc handles explicitly. r=bwc
#   
#   With the libwebrtc update codecs will be created off-main.
#   WebrtcGmpPCHandleSetter not only adds a lot of indirection, it is also
#   main-thread-only. With this patch we can create gmp codecs on any thread.
# ==============================================================================

diff -r 1913c332695e -r 3ca24e761cc3 dom/media/webrtc/libwebrtcglue/VideoConduit.h
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Wed Feb 10 09:58:11 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Wed Feb 10 10:34:14 2021 +0100
@@ -204,11 +204,6 @@
 
   bool HasCodecPluginID(uint64_t aPluginID) override;
 
-  void SetPCHandle(const std::string& aPCHandle) override {
-    MOZ_ASSERT(NS_IsMainThread());
-    mPCHandle = aPCHandle;
-  }
-
   void DeleteStreams() override;
 
   bool Denoising() const { return mDenoising; }
@@ -223,7 +218,8 @@
   }
 
   WebrtcVideoConduit(RefPtr<WebRtcCallWrapper> aCall,
-                     nsCOMPtr<nsISerialEventTarget> aStsThread);
+                     nsCOMPtr<nsISerialEventTarget> aStsThread,
+                     std::string aPCHandle);
   virtual ~WebrtcVideoConduit();
 
   MediaConduitErrorCode InitMain();
@@ -456,9 +452,6 @@
   MediaEventListener mRecvPluginCreated;
   MediaEventListener mRecvPluginReleased;
 
-  // Main thread only
-  std::string mPCHandle;
-
   // Accessed only on mStsThread
   Maybe<DOMHighResTimeStamp> mLastRtcpReceived;
 